File: internal/domain/room.go
package domain

import "time"

type RoomStatus string

const (
	RoomStatusWaiting  RoomStatus = "waiting"
	RoomStatusPlaying  RoomStatus = "playing"
	RoomStatusFinished RoomStatus = "finished"
)

type Room struct {
	ID        int64      `json:"id"`
	Code      string     `json:"code"`
	Mode      GameMode   `json:"mode"`
	Status    RoomStatus `json:"status"`
	CreatedAt time.Time  `json:"created_at"`
}

File: app/domain/room.py
from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel

from app.domain.game import GameMode


class RoomStatus(str, Enum):
    WAITING = "waiting"
    PLAYING = "playing"
    FINISHED = "finished"


class Room(BaseModel):
    id: int | None = None
    code: str
    mode: GameMode
    status: RoomStatus = RoomStatus.WAITING
    created_at: datetime = datetime.now()
